feat(curriculum): New steps and tests for expense tracker (#52570)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-a-expense-tracker/657e6231622cd919e2d7094b.md

---
id: 657e6231622cd919e2d7094b
title: Step 4
challengeType: 20
dashedName: step-4
---

# --description--

> Now it's time to be able to append each expense as a dictionary to the `expenses` list. This illustrates how to create functions that manipulate lists by adding new items.

Reviewer: ```suggestion
Now it's time to be able to append each expense as a dictionary to the `expenses` list. "Appending" means adding an item to the end of the list. 
```

Reviewer: I would suggest adding a brief context on what "appending" means if this is not mentioned in previous steps.

Author: I will commit your suggested change as I think it briefly explains "Appending" without making the step too long.

> To append a value to a list, the `append` methos is used. This is the syntax used to append a value to a list:

Reviewer: ```suggestion
In Python, the `.append()` method is used to append an item to a list. This is an example:
```

> ```python
> def function_name(parameter_name):
>     list_name = []
>     list_name.append(parameter_name)
> ```

Reviewer: ```suggestion

```python
def function_name(parameter_name):
    list_name = []
    list_name.append(parameter_name)
```
```

Reviewer: I would suggest adding a blank line before the example.

> But to append a dictionary to a list, this syntax is used:
> ```python
> def function_name(parameter_name):
>     list_name = []
>     list_name.append({"sample_key": parameter_name})
> ```

Reviewer: ```suggestion
This is an example of how you can append a dictionary to a list:

```python
def function_name(parameter_name):
    list_name = []
    list_name.append({"sample_key": parameter_name})
```
```

> Your task is to give the `add_expense` function the functionality to append a new expense in form of a dictionary to the `expense` list. The dictionary should have two keys, `"amount"` and `"category"`. The values of the keys should be the values of the arguments passed into the `add_expense` function.

Reviewer: ```suggestion
Your task is to give the `add_expense()` function the functionality to append a new expense dictionary to the `expenses` list. The dictionary should have two keys: `"amount"` and `"category"`. The values of the keys should be the values of the arguments passed into the `add_expense` function.
```

# --hints--

> The `add_expense` should use the `append` method to append to the `expense` list

Reviewer: ```suggestion
The `add_expense()` function should call the `.append()` method to append the dictionary to the `expenses` list.
```

> The `add_expense` should use the `append` method to append to the `expense` list
> ```js
> ({
>   test: () => assert.match(code, /expenses\.append\(/)
> })

Reviewer 2: ```suggestion
The `add_expense` should use the `append` method to append to the `expense` list

```js
({
  test: () => assert.match(code, /expenses\.append\(/)
})
```

Reviewer 2: required blank line

Author: Fixed

```
 

> A dictionary containing `amount` and `category` keys should be appended

Reviewer: ```suggestion
You should append a dictionary with the keys `amount` and `category` to the `expenses` list.
```

```js
({

>   test: () => assert.match(code, /expenses\.append\(\{\s*"amount"\s*:\s*amount\s*,\s*"category"\s*:\s*category\s*\}\)/, "The line 'expenses.append({\"amount\": amount, \"category\": category})' should be present in the code.")

Author: Solved
